src/pybind/mgr/cephadm/serve.py

```diff
@@ -993,13 +986,24 @@ def _run_cephadm(self,
                 try:
                     out, err, code = remoto.process.check(
                         conn,
-                        [python, '-u'],
-                        stdin=script.encode('utf-8'))
+                        [python, self.mgr.cephadm_binary_path] + final_args,
+                        stdin=stdin.encode('utf-8') if stdin is not None else None)
+                    if code == 2:
```

Comment:
```suggestion
                    if code == errno.ENOENT:
```

(nit) I was somewhat confused why retval 2, but it's `file not found` from python.